File: easy/1768_merge_string_alternately/solution.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        new_word = []

        for i in range(min(len(word1), len(word2))):
            new_word.extend((word1[i], word2[i]))

        longer_word = word1 if len(word1) > len(word2) else word2
        return "".join(new_word) + longer_word[min(len(word1), len(word2)) :]

File: easy/1768_merge_string_alternately/solution.cpp
#include <string>

class Solution {
public:
  std::string mergeAlternately(std::string word1, std::string word2) {
    const int word1Length = word1.length();
    const int word2Length = word2.length();

    std::string newWord;
    newWord.reserve(word1Length + word2Length);

    int i = 0;

    while (i < word1Length || i < word2Length) {
      if (i < word1Length)
        newWord += word1[i];
      if (i < word2Length)
        newWord += word2[i];
      i++;
    }

    return newWord;
  }
};
